apps/api/src/campaign/campaign.service.ts

```diff
@@ -553,12 +529,17 @@ export class CampaignService {
           },
           select: donationNotificationSelect,
         })
+
+        //if donation is switching to successful, increment the vault amount and send notification
         if (newDonationStatus === DonationStatus.succeeded) {
+          await this.vaultService.incrementVaultAmount(donation.targetVaultId, donation.amount, tx)
```

Comment:
Would there be a problem if this function get's called a second time (another update) and this code block get's re-executed

Reply:
the incrementVaultAmount is executed only on new successful donation or when a waiting donation is turned into successful, so no, there is no way this could be called again on the same donation